src/js/mixins/QueryParamsMixin.jsx

```diff
@@ -79,6 +79,10 @@ var QueryParamsMixin = {
     }
 
     router.transitionTo(router.getCurrentPathname(), {}, queryParams);
+  },
+
+  decodeArray: function (array) {
```

Comment:
Not sure. This function does not handle decoding a generic array, but an array of labels expressed as key:value pairs in the query params. It may sound misleading, as if we are using this to decode ALL "array-like" values. 
Imho we should at least convey that it's currently used for this specific case. Maybe `decodeLabelsQueryParam`?